src/main/java/interface_adapters/tui/controllers/RecipeTransferOperation.java

```diff
@@ -0,0 +1,140 @@
+package interface_adapters.tui.controllers;
+
+import database.Database;
+import entity.Recipe;
+import entity.RecipeData;
+import interface_adapters.tui.Colour;
+import interface_adapters.tui.TextualOperation;
+import interface_adapters.tui.TextualReader;
+import usecase.RecipeDataConverter;
+import usecase.RecipeManager;
+
+import javax.swing.*;
+import java.io.*;
+import java.util.List;
+
+/**
+ * An operation for handling the use case of importing and exporting recipes.
+ */
+public class RecipeTransferOperation implements TextualOperation {
+    private final TextualReader reader;
+    private final RecipeDataConverter converter;
+    private final RecipeManager recipeManager;
+
+    private final List<TextualOperation> operations = List.of(
+            new RecipeImporter(),
+            new RecipeExporter()
+    );
+
+    public RecipeTransferOperation(TextualReader reader, Database database) {
```

Comment:
Top level operations shouldn't be passed the database in their constructor, they should operate independently of any database at all. I would recommend passing the recipe manager and recipe data converter use-cases directly to the constructor instead of the database.